src/smhi/strang.py

```diff
@@ -1,8 +1,309 @@
 """
 SMHI STRÅNG client.
+
+See validation of model: https://strang.smhi.se/validation/validation.html
 """
+import json
+import arrow
+import requests
+import logging
+from datetime import datetime
+from functools import partial
+from collections import defaultdict
+from requests.structures import CaseInsensitiveDict
+from smhi.constants import (
+    STRANG,
+    STRANG_POINT_URL,
+    STRANG_PARAMETERS,
+    STRANG_MULTIPOINT_URL,
+    STRANG_TIME_INTERVALS,
+)
+
+
+logging.basicConfig(level=logging.INFO)
 
 
 class Strang:
-    def __init__():
-        pass
+    """
+    SMHI STRÅNG class. Only supports category strang1g and version 1.
+    """
+
+    def __init__(self) -> None:
+        """
+        Initialise STRÅNG object.
+        """
+        self._category = "strang1g"
+        self._version = 1
+
+        self.latitude = None
+        self.longitude = None
+        self.parameter = None
+        self.time_from = None
+        self.time_to = None
+        self.valid_time = None
+        self.time_interval = None
+        self.status = None
+        self.header = None
+        self.data = None
+        self.available_parameters = STRANG_PARAMETERS
+        self.point_raw_url = partial(
+            STRANG_POINT_URL.format, category=self._category, version=self._version
+        )
+        self.multipoint_raw_url = partial(
+            STRANG_MULTIPOINT_URL.format, category=self._category, version=self._version
+        )
+        self.point_url = None
+        self.multipoint_url = None
+
+    @property
+    def parameters(self) -> defaultdict:
+        """
+        Get parameters property.
+
+        Returns:
+            available parameters
+        """
+        return self.available_parameters
+
+    def get_point(
+        self,
+        longitude: float,
+        latitude: float,
+        parameter: int,
+        time_from: str = None,
+        time_to: str = None,
+        time_interval: str = None,
+    ) -> tuple[bool, CaseInsensitiveDict, list]:
+        """
+        Get data for given lon, lat and parameter.
+
+        Args:
+            longitude: longitude
```

Comment:
It may be somewhat more intuitive to have the latitude before the longitude in the calls as I think that is the idiomatic way of writing.

Reply:
Agree, should be fixed in 709811b.